shared/scripts/virtio_console_guest.py

```diff
@@ -902,18 +902,18 @@ def send(self, port, length=1, mode=True, is_static=False):
 
         if not is_static:
             while len(data) < length:
-                data += "%c" % random.randrange(255)
+                data += "%s" % random.randrange(255)
```

Comment:
Well, this change is incorrect, it wants to get length of bytestring data here not textstring data.
```python
Python 2.7.15 (default, May 16 2018, 17:50:09) 
[GCC 8.1.1 20180502 (Red Hat 8.1.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> "%c" % 128
'\x80'
>>> "%s" % 128
'128'
```
It should be:
```python
data = b""
...
        data += b"%c" % random.randrange(255)
```

Reply:
@luckyh With "%c", it send these bytes directly, I don't find a way to decode it after receive the data from socket. That' s why I use string and encode it before sending. b"%c" does not resolve this, still can not decode it after received:

qemu/tests/virtio_console.py 

if (int(port.sock.recv(1024)) < 10):
Will hit error:
```
04:52:34 ERROR| ValueError: invalid literal for int() with base 10: b'P}\xcdm\xc0f\xeb\x8d*\xf3'
```
With decode():
if (int(port.sock.recv(1024).decode()) < 10):
```
L0984 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/virtio_console.py", line 255, in test_rw_host_offline
    if (int(port.sock.recv(1024).decode()) < 10):

2018-08-30 04:44:27,277 test             L0984 ERROR| UnicodeDecodeError: 'utf-8' codec can't decode byte 0x99 in position 0: invalid start byte
```
Any suggestions? Thanks.

Reply:
And with current "%s" + encode() way, what I received is:
```
05:39:35 DEBUG| [stdout] b'13739159193'
```
I suppose that should be what intended.

Reply:
It is exactly a bug of `tp-qemu/qemu/tests/virtio_console.py`.
```python
    def test_rw_host_offline():
        """
        Try to read from/write to host on guest when host is disconnected.
        :param cfg: virtio_console_params - which type of virtio port to test
        :param cfg: virtio_port_spread - how many devices per virt pci (0=all)
        """
        (vm, guest_worker, port) = virtio_test.get_vm_with_single_port(
            params.get('virtio_console_params'))
        if port.is_open():
            port.close()

        guest_worker.cmd("virt.recv('%s', 0, 1024, False)" % port.name, 10)
        match, tmp = guest_worker._cmd("virt.send('%s', 10, True)" % port.name,
                                       10)
        #                            send 10 characters ^^
        if match is not None:
            test.fail("Write on guest while host disconnected "
                      "didn't time out.\nOutput:\n%s" % tmp)

        port.open()

        if (port.sock.recv(1024) < 10):
        # why compare the content of the received data with integer number '10'?
        # should be `len(port.sock.recv(1024)) < 10`, right?
```

Reply:
@luckyh You are right, thanks for you patience, I have updated the code and add you to sign off list. Please help review again.